File: MODULE_4/weekly_assignment/customerproduct/Ecom/src/main/java/org/northernarc/ecom/service/OrderService.java
package org.northernarc.ecom.service;

import org.northernarc.ecom.DTO.OrderRequestDTO;
import org.northernarc.ecom.DTO.OrderResponseDTO;
import org.northernarc.ecom.DTO.OrderUpdateDTO;

import java.util.List;

public interface OrderService {

    OrderResponseDTO placeOrder(OrderRequestDTO request);

    List<OrderResponseDTO> getAllOrders();

    OrderResponseDTO getOrderById(Integer id);

    void cancelOrder(Integer orderId);

    List<OrderResponseDTO> getOrdersByCustomer(Integer customerId);

    OrderResponseDTO updateOrder(Integer id,
                                 OrderUpdateDTO dto);
}

File: MODULE_6/day5/EcomFrontEndDev/src/app/services/order.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { OrderRequestDTO, OrderResponseDTO, OrderUpdateDTO } from '../dto/order';

@Injectable({
	providedIn: 'root'
})
export class OrderService {
	private readonly baseUrl = 'http://localhost:8080/api/ecom/order';

	constructor(private readonly http: HttpClient) {}

	getAllOrders(): Observable<OrderResponseDTO[]> {
		return this.http.get<OrderResponseDTO[]>(this.baseUrl);
	}

	getOrderById(id: number): Observable<OrderResponseDTO> {
		return this.http.get<OrderResponseDTO>(`${this.baseUrl}/${id}`);
	}

	placeOrder(payload: OrderRequestDTO): Observable<OrderResponseDTO> {
		return this.http.post<OrderResponseDTO>(this.baseUrl, payload);
	}

	updateOrder(id: number, payload: OrderUpdateDTO): Observable<OrderResponseDTO> {
		return this.http.put<OrderResponseDTO>(`${this.baseUrl}/${id}`, payload);
	}

	cancelOrder(id: number): Observable<void> {
		return this.http.delete<void>(`${this.baseUrl}/${id}`);
	}

	getOrdersByCustomer(id: number): Observable<OrderResponseDTO[]> {
		return this.http.get<OrderResponseDTO[]>(`${this.baseUrl}/customer/${id}`);
	}
}
